### ISSUE TITLE
[CRASH] [forge]ctov-3.4.10 CAUSING GAME TO CRASH EVERY 15 MINUTES. GAME IS UNPLAYABLE. FORCED TO UNINSTALL MOD TO PLAY AGAIN. RELATED TO "FIXED" OLD BUG: ATM [Crash]: Worldgen crash #1497 

### ISSUE THREAD

**Issue by Minebrah** [ChoiceTheorem/ChoiceTheorem-s-overhauled-village#233]
**Describe the bug**
I will just be exploring and my game will crash suddenly. It happens after playing about 15 minutes or so and it alwasy happens making the game unplayable. I looked at the logs and it looks like its multiple problems with this mod. at least one of which looks to have been brought up in the past and was thought to be fixed.

[21:04:27] [Server thread/ERROR] [minecraft/BlockableEventLoop]: Error executing task on Chunk source main thread executor for minecraft:overworld
net.minecraft.ReportedException: Feature placement

-- Feature --
Details:
	Description: ResourceKey[minecraft:worldgen/structure / ctov:small/village_desert_oasis]

Caused by: java.lang.RuntimeException: com.mojang.brigadier.exceptions.CommandSyntaxException: Unknown block type 'rats:garbage_pile' at position 0: <--[HERE]
	at net.minecraft.world.level.levelgen.structure.templatesystem.JigsawReplacementProcessor.m_7382_(JigsawReplacementProcessor.java:46) ~[client-1.20.1-20230612.114412-srg.jar%231652!/:?] {re:classloading}

My Crash Report 1:

https://[gist.githubusercontent.com/Minebrah/701bc875691aac0d2c38736c25b644b0/raw/97b08b23574b18dac50c76d75ec178c186fa6365/gistfile1.txt](https://gist.githubusercontent.com/Minebrah/701bc875691aac0d2c38736c25b644b0/raw/97b08b23574b18dac50c76d75ec178c186fa6365/gistfile1.txt)

[https://[gist.githubusercontent.com/Minebrah/701bc875691aac0d2c38736c25b644b0/raw/97b08b23574b18dac50c76d75ec178c186fa6365/gistfile1.txt](https://gist.githubusercontent.com/Minebrah/701bc875691aac0d2c38736c25b644b0/raw/97b08b23574b18dac50c76d75ec178c186fa6365/gistfile1.txt)](url)

My Crash report 2:

https://gist.github.com/Minebrah/72d247d84bf85061d53704e4711e7c0f
[https://gist.github.com/Minebrah/72d247d84bf85061d53704e4711e7c0f](url)

Link to old CTOV issue report that is similar to mine:

https://github.com/AllTheMods/ATM-9/issues/1497
[https://github.com/AllTheMods/ATM-9/issues/1497](url)

**Minecraft Version**
1.20.1 

**CTOV Version**
[forge]ctov-3.4.10
**Add-ons used (Before 1.20)**

**Do you install Lithostitched?(1.20+)**
Yes

**Additional context**

**Comment by Korenchkin** [comment]
oh yeah,rats:garbage_pile,i have this too on world generation crash..latest now 1.20.1

**Comment by Korenchkin** [comment]
[https://github.com/ChoiceTheorem/ChoiceTheorem-s-overhauled-village/issues/91](url) same issue

**Comment by ChoiceTheorem** [comment]
This bug comes back from the dead huh?

**Comment by Minebrah** [comment]
yup. It's reared its naked face I'm afraid. like I mentioned the mod is unplayable on 1.20.1 forge at the moment. a fix would be greatly appreciated.

**Comment by ChoiceTheorem** [comment]
I'll scramble for another fix this week

**Comment by ChoiceTheorem** [comment]
1. Remove all add-ons first.
2. I recheck the code. This bug shouldn't be possible now. I need you to do half by half disable the mod until you can find me the co-culprit with CTOV as this is not CTOV alone.

**Comment by JacobTDC** [comment]
I'm also experiencing this issue. Trying to narrow down a mod right now.

**Comment by JacobTDC** [comment]
So, in my case, it appears to have been an incompatibility with a mod called Cyanide.

**Comment by xXsansoloXx** [comment]
> So, in my case, it appears to have been an incompatibility with a mod called Cyanide.

This shouldn't be possible. Cyanide is used for stack tracing in crash reports and logs. It just shows its name there. Regardless, i'm going to test right now because I'm experiencing this exact issue with the croptopia and friends and foes addons.

Previously (before this test) I also had the Villager's Plus add on. Just speculation, but it might be that. I'll reply shortly with my findings.

**Comment by JacobTDC** [comment]
> > So, in my case, it appears to have been an incompatibility with a mod called Cyanide.
> 
> This shouldn't be possible. Cyanide is used for stack tracing in crash reports and logs. It just shows its name there. Regardless, i'm going to test right now because I'm experiencing this exact issue with the croptopia and friends and foes addons.
> 
> Previously (before this test) I also had the Villager's Plus add on. Just speculation, but it might be that. I'll reply shortly with my findings.

I thought Cyanide couldn't possibly be the culprit, either, but when I removed it, it inexplicably started working again. And it wasn't giving me Cyanide errors, it was giving me a regular Java stack trace.

But I was getting the same error as OP, the one about 'rats:garbage_pile'.

**Comment by xXsansoloXx** [comment]
Update: It was indeed the Villager's Plus Add on

**Comment by xXsansoloXx** [comment]
> > > So, in my case, it appears to have been an incompatibility with a mod called Cyanide.
> > 
> > 
> > This shouldn't be possible. Cyanide is used for stack tracing in crash reports and logs. It just shows its name there. Regardless, i'm going to test right now because I'm experiencing this exact issue with the croptopia and friends and foes addons.
> > Previously (before this test) I also had the Villager's Plus add on. Just speculation, but it might be that. I'll reply shortly with my findings.
> 
> I thought Cyanide couldn't possibly be the culprit, either, but when I removed it, it inexplicably started working again. And it wasn't giving me Cyanide errors, it was giving me a regular Java stack trace.
> 
> But I was getting the same error as OP, the one about 'rats:garbage_pile'.

That's very strange... ChoiceTheorum what is going on ;-; this mod scares me

**Comment by xXsansoloXx** [comment]
Update 2: It was indeed not the Villager's Plus Add on

**Comment by xXsansoloXx** [comment]
> Update 2: It was indeed not the Villager's Plus Add on

I just got lucky, and something else crashed it (I'm making my modpack, and havent tested it up until this point :P )

**Comment by ChoiceTheorem** [comment]
Why modpack creators insists on using add-on that is no longer needed?

**Comment by Tempest-5ky** [comment]
Also having issues with this! I don't have any of the addons installed, just CTOV. Crashes the server very frequently during worldgen :( 
Like most people here, it's specifically an issue with a missing block from 'rats', a mod I do not have installed.

**Comment by ChoiceTheorem** [comment]
Right now, I'm suspecting some bugs or mistake with lithostitched

**Comment by Alvareto136** [comment]
I've also been having this problem on CTOV 3.4.12 (forge 1.20.1). The crash log mentions Lithostiched, (CTOV dependency), Alex Caves, Citadel (Alex Caves dependency), Cyanide, ServerCore and ModernFix.

I'll leave my crash log attached, it may be useful.

[crash-2025-03-05_21.32.19-server.txt](https://github.com/user-attachments/files/19100487/crash-2025-03-05_21.32.19-server.txt)

**Comment by ChoiceTheorem** [comment]
Try uninstalling cyanide first as currently, there's cyanide-lithostitched
conflict.

**Comment by Alvareto136** [comment]
Hi, I recently got the crash again, but this time there was no Cyanide mod to blame, just a bunch of mixins stuff:

- [Crash without Cyanide](https://github.com/user-attachments/files/19513852/crash-2025-03-28_12.38.02-server.txt)

So, I tested if just CTOV, Lithostiched and Cyanide would crash in Forge and Fabric, and it did:

- [Crash in both Forge and Fabric](https://paste.atlauncher.com/view/6f791bfa-1bec-4dcb-9018-e16ff854520b)
